components/MapComponent.tsx

```diff
@@ -1,123 +1,146 @@
-import React, { useState } from "react";
-import { Button, StyleProp, View, ViewStyle } from "react-native";
-import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";
+import { Location } from "lib/location/Location"
+import React, { useState } from "react"
+import { StyleProp, View, ViewStyle } from "react-native"
+import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps"
+
+// Type for the map markers, in order to separate their logic out
+export interface mapMarker {
+  key: string
+  title: string
+  location: Location
+}
+
+interface Props<T> {
+  // Contain style specifically for style on container for map
+  // Map style specifically for style on map
+  containStyle?: StyleProp<ViewStyle>
+  mapStyle?: StyleProp<ViewStyle>
 
-interface Props {
-  containStyle?: StyleProp<ViewStyle>;
-  mapStyle?: StyleProp<ViewStyle>;
   initialRegion: {
```

Comment:
InitialRegion can be ripped straight from the MapProps of the package by importing MapProps from the package and making the type like “MapProps[“initialRegion”]”. Idk if MapProps is the exact name of the type but you get the idea